doc/source/development/contributing_codebase.rst

```diff
@@ -260,9 +260,9 @@ pandas uses `mypy <http://mypy-lang.org>`_ and `pyright <https://github.com/micr
 
 .. code-block:: shell
 
-   ./ci/code_checks.sh typing
+   pre-commit --hook-stage manual --all-files
```

Comment:
I tried this but got
```
$ pre-commit --hook-stage manual --all-files
usage: pre-commit [-h] [-V]
                  {autoupdate,clean,hook-impl,gc,init-templatedir,install,install-hooks,migrate-config,run,sample-config,try-repo,uninstall,help}
                  ...
pre-commit: error: argument command: invalid choice: 'manual' (choose from 'autoupdate', 'clean', 'hook-impl', 'gc', 'init-templatedir', 'install', 'install-hooks', 'migrate-config', 'run', 'sample-config', 'try-repo', 'uninstall', 'help')
```

Does something need doing before that?

Reply:
should have been 

> pre-commit **run** --hook-stage manual --all-files

Reply:
getting another error now
```
$ pre-commit run --hook-stage manual --all-files
An error has occurred: InvalidConfigError: 
==> File .pre-commit-config.yaml
==> At Config()
==> At key: repos
==> At Repository(repo='https://github.com/pre-commit/pre-commit-hooks')
==> At key: hooks
==> At Hook(id='end-of-file-fixer')
==> At key: stages
==> At index 8
=====> Expected one of commit, commit-msg, manual, merge-commit, post-checkout, post-commit, post-merge, prepare-commit-msg, push but got: 'post-rewrite'
Check the log at /home/marco/.cache/pre-commit/pre-commit.log
```

Reply:
looks like `post-rewrite` is a newer one. Doesn't look like it's in use anywhere here anyway? I'd suggest to either leave it out, or to bump the minimum pre-commit version to be higher than 2.9.2

Reply:
Require pre-commit 2.15.0 now (the oldest version that supports this stage)